### ISSUE TITLE
'space-before-function-paren' returns error/warning for functions defined as object attributes

### ISSUE THREAD

**Issue by darrenklein** [eslint/eslint#9100]
**Tell us about your environment**

* **ESLint Version:**
4.4.1
* **Node Version:**
8.2.1
* **npm Version:**
5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** 'babel-eslint'

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	parser: "babel-eslint",
	"plugins": [
		"react",
		"babel"
	],
	"parserOptions": {
		"ecmaFeatures": {
			"jsx": true
		},
		"ecmaVersion": 6,
		"sourceType": "module"
	},
	"rules": {
		"array-element-newline": 0,
		"arrow-body-style": 0,
		"arrow-parens": [0, "as-needed"],
		"capitalized-comments": 0,
		"dot-location": [0, "property"],
		"eol-last": 0,
		"func-names": [0, "as-needed"],
		"id-length": 0,
		"indent": 0,
		"init-declarations": 0,
		"line-comment-position": 0,
		"max-len": 0,
		"max-statements": 0,
		"newline-per-chained-call": 0,
		"no-alert": 0,
		"no-extra-parens": [1, "all", { ignoreJSX: "all" }],
		"no-inline-comments": 0,
		"no-magic-numbers": [1, {"ignoreArrayIndexes": true, "ignore": [1]}],
		"no-mixed-spaces-and-tabs": [0, "smart-tabs"],
		"no-plusplus": [0, {"allowForLoopAfterthoughts": true}],
		"no-process-env": 0,
		"no-sync": 0,
		"no-tabs": 0,
		"object-curly-newline": 0,
		"object-curly-spacing": 0,
		"object-shorthand": 0,
		"padded-blocks": 0,
		"quote-props": 1,
		"react/display-name": 0,
		"react/jsx-closing-bracket-location": [1, "after-props"],
		"react/jsx-handler-names": 0,
		"react/jsx-indent": [1, "tab"],
		"react/jsx-indent-props": [1, "tab"],
		"react/jsx-max-props-per-line": 0,
		"react/jsx-no-literals": 0,
		"react/jsx-sort-props": 0,
		"semi": 0,
		"sort-imports": 0,
		"sort-keys": 0,
		"sort-vars": 0,
		"space-before-function-paren": [1, "never"]
	},
	"globals": {
		"alert": true,
		"API_URL": true,
		"ExtractTextPlugin": true,
		"fetch": true,
		"it": true,
		"localStorage": true,
		"Promise": true
	},
	"extends": ["eslint:all", "plugin:react/all"],
	"env": {
		"es6": true,
		"browser": true,
		"node": true
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const errorProcessing = {
	"processErrorOnForm": function(result, reactThis) {
		if (result.response_not_ok) {
			reactThis.setState({
				"error": Object.assign(result, {"isError": true})
			})
		} else {
			reactThis.setState({
				"error": {
					"errors": result.errors,
					"isError": true,
					"message": result.message
				}
			})
		}
	}
};

export default errorProcessing;
```

**What did you expect to happen?**
I expected that this code would not throw any errors or warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
Despite the fact that there is no space before my function parentheses, the following warning was returned:

`2:32 warning Unexpected space before function parentheses space-before-function-paren`

**Comment by platinumazure** [comment]
Can you reproduce this with the default parser?

**Comment by darrenklein** [comment]
I changed the parser options to

```
	"parserOptions": {
		"ecmaVersion": 5,
		"sourceType": "script"
	},
```
which I believe are the default settings - the warning went away.

**Comment by not-an-aardvark** [comment]
Thanks for the response. I think this might be an issue with the [`babel-eslint`](https://github.com/babel/babel-eslint) parser, so feel free to report an issue there. However, since the issue doesn't occur with the default parser, I don't think it's a problem with ESLint, so I'm going to close this.
